File: Latihan2-cpp/Memory.cpp
#include "Hardware.cpp"

using namespace std;

class Memory : public Hardware{

    private:

    string frequency;
    string memorySize;
    string supportsCuda;

    public:

    // Constructor
    Memory(){
    }

    Memory(string frequency, string memorySize, string supportsCuda){
        this->frequency = frequency;
        this->memorySize = memorySize;
        this->supportsCuda = supportsCuda;
    }

    // Get-set Frequency
    void setFrequency(string frequency){
        this->frequency = frequency;
    }

    string getFrequency(){
        return this->frequency;
    }

    // Get-set Memory Size
    void setMemorySize(string memorySize){
        this->memorySize = memorySize;
    }

    string getMemorySize(){
        return this->memorySize;
    }

    // Get-set Supports Cuda
    void setSupportsCuda(string supportsCuda){
        this->supportsCuda = supportsCuda;
    }

    string getSupportsCuda(){
        return this->supportsCuda;
    }

    ~Memory(){
    }

};

File: Latihan2-py/Memory.py
from Hardware import Hardware # Superclass

class Memory(Hardware):
    # Constructor
    def __init__(self):
        self.__frequency = ""
        self.__memorySize = ""
        self.__supportsCuda = ""
    
    # Get-set Frequency
    def setFrequency(self, frequency):
        self.__frequency = frequency

    def getFrequency(self):
        return self.__frequency
    
    # Get-set MemorySize
    def setMemorySize(self, memorySize):
        self.__memorySize = memorySize

    def getMemorySize(self):
        return self.__memorySize
    
    # Get-set SupportsCuda
    def setSupportsCuda(self, supportsCuda):
        self.__supportsCuda = supportsCuda

    def getSupportsCuda(self):
        return self.__supportsCuda